src/compute-pr-actions.ts

```diff
@@ -181,15 +186,21 @@ function extendPrInfo(info: PrInfo): ExtendedPrInfo {
     const hasChangereqs = changereqReviews.length > 0;
     const approvalFlags = getApprovalFlags();
     const approverKind = getApproverKind();
-    const approved = getApproval();
+    const firstApprovalDate = hasChangereqs ? undefined : earliestDate(...approvedReviews.map(r =>
+        r.isMaintainer
+        || approverKind === "owners" && allOwners.includes(r.reviewer)
+        || approverKind === "others"
+            ? r.date : undefined));
+    const approved = !!firstApprovalDate;
     const canBeMerged = info.ciResult === CIResult.Pass && !info.hasMergeConflict && approved;;
+    const mergeIsRequested = !!info.lastMergeRequestDate && !!firstApprovalDate && info.lastMergeRequestDate > latestDate(info.lastPushDate, info.reopenedDate, firstApprovalDate)!;

```

Comment:
This compares dates by converting them to strings.

Reply:
I think they're converted [to numbers](https://tc39.es/ecma262/#_ref_1996)? e.g. `String(new Date(2020, 10, 15)) > String(new Date(2020, 10, 16))` but `new Date(2020, 10, 15) < new Date(2020, 10, 16)`

Reply:
Ah, thanks for finding that reference.  (I suspected that it might use iso strings for the comparison, looks like it doesn't.)  I'll simplify other places too.